common/djangoapps/course_modes/models.py

```diff
@@ -375,6 +403,128 @@ def min_course_price_for_currency(cls, course_id, currency):
+        from student.helpers import (
+            VERIFY_STATUS_NEED_TO_VERIFY,
+            VERIFY_STATUS_SUBMITTED,
+            VERIFY_STATUS_APPROVED
+        )
+
+        show_image = False
+        enrollment_title, enrollment_value, image_alt = '', '', ''
+
+        if mode == "verified":
+            if verification_status in [VERIFY_STATUS_NEED_TO_VERIFY, VERIFY_STATUS_SUBMITTED]:
+                enrollment_title = __("Your verification is pending")
+                enrollment_value = __("Verified: Pending Verification")
+                show_image = True
+                image_alt = __("ID verification pending")
+            elif verification_status == VERIFY_STATUS_APPROVED:
+                enrollment_title = __("You're enrolled as a verified student")
+                enrollment_value = __("Verified")
+                show_image = True
+                image_alt = __("ID Verified Ribbon/Badge")
+            else:
+                enrollment_title = __("You're enrolled as an honor code student")
+                enrollment_value = __("Honor Code")
+        elif mode == "honor":
+            enrollment_title = __("You're enrolled as an honor code student")
+            enrollment_value = __("Honor Code")
+        elif mode == "audit":
+            enrollment_title = __("You're auditing this course")
+            enrollment_value = __("Auditing")
+        elif mode in["professional", "no-id-professional"]:
+            enrollment_title = __("You're enrolled as a professional education student")
+            enrollment_value = __("Professional Ed")
+
+        return {
+            'enrollment_title': enrollment_title,
+            'enrollment_value': enrollment_value,
+            'show_image': show_image,
+            'image_alt': image_alt,
+            'display_mode': CourseMode._get_certificate_display_mode(mode, verification_status)
+        }
+
+    @staticmethod
+    def _get_certificate_display_mode(enrollment_mode, verification_status):
+        """
+         return the of mode after checking enrollment mode and status.
+         Args:
+            mode (str): enrollment mode.
+            verification_status (str) : verification status of student
+
+        Returns:
+            str: display mode for certs
+        """
+
+        # import inside the function to avoid the circular import
+        from student.helpers import (
+            VERIFY_STATUS_NEED_TO_VERIFY,
+            VERIFY_STATUS_SUBMITTED,
+            VERIFY_STATUS_APPROVED
+        )
+
+        if enrollment_mode == "verified":
+            if verification_status in [VERIFY_STATUS_NEED_TO_VERIFY, VERIFY_STATUS_SUBMITTED, VERIFY_STATUS_APPROVED]:
+                mode = "verified"
+            else:
+                mode = "honor"
+        elif enrollment_mode == "professional" or enrollment_mode == "no-id-professional":
+            mode = "professional"
+        else:
+            mode = enrollment_mode
```

Comment:
In case of else it will return the enrollment.mode value. its old functionality so i maintained that.
https://github.com/edx/edx-platform/blob/master/lms/templates/dashboard/_dashboard_course_listing.html#L40